Tested #23622 -- ordering retained for distinct on fields subqueries

akaariai opened the pull request:
The ticket was already fixed by
b68212f539f206679580afbfd008e7d329c9cd31.

Thanks to Beauhurst for commissioning the work on this ticket.

timgraham commented on tests/queries/tests.py:
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
+        a1.save()
+        a2 = Ticket23605A()
+        a2.save()
+        c1 = Ticket23605C.objects.create(field_c0=0.0)
```
Is there a reason to mix `create()` and `save()`?

timgraham commented on tests/queries/tests.py:
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
+        Ticket23605B(
+            modela_fk=a1, field_b0=123,
+            field_b1=datetime.date(2013, 1, 6),
+            modelc_fk=c1
```
add trailing comma on kwargs

timgraham commented on tests/queries/tests.py:
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
+        # Make sure __pk__in and __in work the same for related fields when
+        # using a distinct on subquery.
+        qx = (
+            Q(ticket23605b__pk__in=Ticket23605B.objects.order_by("modela_fk", "-field_b1").distinct("modela_fk"))
```
use single quotes

timgraham commented on tests/queries/tests.py:
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
+            Q(ticket23605b__in=Ticket23605B.objects.order_by("modela_fk", "-field_b1").distinct("modela_fk"))
+            & Q(ticket23605b__field_b0__gte=50)
+        )
+        self.assertEqual(
```
do you want to add an assertion for the results of the queryset to assure that doesn't change?

timgraham commented on tests/queries/tests.py:
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
+            modelc_fk=c1
+        ).save()
+
+        # Make sure __pk__in and __in work the same for related fields when
```
this could be the docstring so it's more clear upfront what the purpose of the test is?

akaariai commented:
I fixed the review comments. I also fixed a test that had assumptions about the pk values in separate commit.

I quickly checked both PEP-8 and our style guide and didn't see where we recommend trailing commas for kwargs or use of single quote characters vs double quote characters. If we have such rules, fine, but if not I'm not that enthusiastic about adding such rules. Especially if we can't automatically check them.

timgraham commented:
merged in 6a9a9e50f233b3ba4af13bf2dc664dfa3b12038d and baa732ac9f18b6ed0ca386c632e15c9fb3037e39, thanks!
